### PR TITLE
[qt] Use the general search-in-downloader machinery.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: qt/update_dialog.cpp [hunk 1]
```diff
@@ -402,8 +407,43 @@ namespace qt
...
+        if (results.m_endMarker)
+        {
+          done = true;
+          cv.notify_one();
+        }
+      };
+
+      m_framework.SearchInDownloader(params);
+
+      {
```
Comment (ygorshenin) [inline qt/update_dialog.cpp]: I generally don't like the idea of blocking call to Search. Could you please investigate - is it possible to fill results asynchronously?

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

**Comment by mapsmetest** [comment]
Can one of the admins verify this patch?

File: qt/update_dialog.cpp (hunk 1 above)
Comment (mpimenov) [reply qt/update_dialog.cpp]: done

**Comment by tatiana-yan** [comment]
насколько сложно сделать ранжирование в соответствии с тем, что отдает поиск?

File: qt/update_dialog.cpp [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 #include "base/assert.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/string_utils.hpp"
 
-#include "std/algorithm.hpp"
-#include "std/bind.hpp"
+#include <algorithm>
+#include <condition_variable>
+#include <functional>
+#include <mutex>
```
Comment (tatiana-yan) [inline qt/update_dialog.cpp]: condition_variable и mutex ещё нужны?

Comment (mpimenov) [reply qt/update_dialog.cpp]: Нет.
done

File: qt/update_dialog.cpp [hunk 3]
```diff
@@ -196,13 +225,140 @@ namespace qt
     done(0);
   }
 
-  void UpdateDialog::OnTextChanged(QString const & text)
+  void UpdateDialog::OnLocaleTextChanged(QString const & text)
+  {
+    string locale(text.toUtf8().constData());
+    strings::Trim(locale);
+    m_locale = std::move(locale);
```
Comment (ygorshenin) [inline qt/update_dialog.cpp]: Why not:
```c++
m_locale.assign(text.toUtf8().constData());
strings::Trim(locale);
```
Not a big deal, in any case.

File: qt/update_dialog.cpp [hunk 4]
```diff
@@ -196,13 +225,140 @@ namespace qt
...
+
+    RefillTree();
+  }
+
+  void UpdateDialog::OnQueryTextChanged(QString const & text)
+  {
+    m_query = string(text.toUtf8().constData());
```
Comment (ygorshenin) [inline qt/update_dialog.cpp]: optional: use assign.

File: qt/update_dialog.cpp (hunk 3 above)
Comment (mpimenov) [reply qt/update_dialog.cpp]: done

File: qt/update_dialog.cpp (hunk 4 above)
Comment (mpimenov) [reply qt/update_dialog.cpp]: done

**Pull request merged**